tests/integration_tests/functional/test_rng.py

```diff
@@ -84,7 +84,7 @@ def _get_percentage_difference(measured, base):
     if measured == base:
         return 0
     try:
-        return (abs(measured - base) / base) * 100.0
+        return ((measured - base) / base) * 100.0
```

Comment:
why do we need the 10% margin actually? Should the measured value not be less or equal to the desired?

Reply:
you are right! So, my experiments show that some times we overshoot. See here for example where I reduced the limit to 1%: https://buildkite.com/firecracker/firecracker-pr/builds/10042

We seem to be getting 102KB/sec when the limit is 100KB/sec. So there is some margin there, which is obviously smaller than 10%. I think, I initially had 10% there because that's the kind of failures I was seeing on the other way of the comparison (90KB/sec), so I added the same range both ways.

In fact, now I changed this to 5% just to avoid intermittent issues.

Does that make sense to you?

Reply:
What is the intuition behind this excess? Does it look like we genuinely allow for slightly more traffic than the limiter is supposed to enforce? Or is this some kind of a rounding error?
I think it would be nice to find a feasible explanation and leave a comment in the test about it.
I don't want to block the PR due to this, because the change is an improvement regardless.

Reply:
I think one source is that the bucket starts off as full. So within the first time delta ε, we get an instantaneous theoretical throughput of (bucket size) / ε, and only after that do we hit (bucket size)/(refill time). The effect of this will be larger the smaller ε is, and is then amortized over the runtime of the test. 

We can test this theory by increasing the runtime of the test - the longer the test, the more diluted this initial effect should be.

Reply:
That sounds plausible indeed. Would that be sensible to obtain more than one datapoint in the test (a long and a short one) to confirm that? Also since we know the duration of the test and the depth of the bucket, can we calculate the margin based on that instead of picking it randomly/empirically?

Reply:
So we do `size * 100` many random bytes, which means `size` random bytes are read immediately, and then `size * 99` random bytes are read at a rate of `size / refill_time`. So in total the test runtime should be `99 * refill_time`, so the throughput will be `size * 100 / (99 * refill_time) = (100 / 99) * (size / refill_time) = (100/99) * expected_throughput_rate`. 100 / 99 is 1.01(01), so 1.01(01)% excess. So I'd say 2% should be a safe enough threshold to account for rounding/measurement errors

Reply:
20 subsequent runs on all instance types and kernel combinations pass successfully, for a limit of 2%: https://buildkite.com/firecracker/bchalios-ad-hoc-tests/builds/42#_.

I pushed a version which adds @roypat's math in the comment. @kalyazin @pb8o could you take a look when you have time?